### PR TITLE
fix(Node.js request strategy): differentiate connect from complete timeout

### PR DESCRIPTION
**Pull request by vvo**
Before this commit, Node.js implementation had only a complete timeout
of 30s. Without any connect timeout.

Now Node.js has a 2s connect timeout, 5s read timeout, 30s write
timeout as per specifications, cc @asurak

There's also an opportunity to add connect timeout handling to
frontend users. Should we? If so, what would be a good connect timeout
knowing that we already have lowered the "read" timeout for instance.

### CODE REVIEW

File: src/AlgoliaSearchCore.js [hunk 1]
```diff
@@ -686,6 +710,13 @@ AlgoliaSearchCore.prototype._incrementCurrentHostIndex = function(hostType) {
   );
 };
 
+AlgoliaSearchCore.prototype._getTimeoutsForRequest = function(hostType, tries) {
+  return {
+    connect: this._timeouts.connect * tries,
```
Comment (bobylito) [inline src/AlgoliaSearchCore.js]: To my knowledge, timeouts gets double after each try. Here we have timeouts that are progressing linearly. Is it on purpose?

More details on what I understand if the connect timeout is 1:
 - here for each tries (3): 1, 2, 3
 - what I thought is "sorta" standard: 1, 2, 4

*update after talking with @clement-leprovost* It is the same in the swift counterpart :)

File: src/reactnative/builds/algoliasearch.reactnative.js [hunk 2]
```diff
@@ -24,6 +24,12 @@ function algoliasearch(applicationID, apiKey, opts) {
 
   opts._ua = opts._ua || algoliasearch.ua;
 
+  opts.timeouts = opts.timeouts || {
+    connect: 2 * 1000,
+    read: 4 * 1000,
```
Comment (bobylito) [inline src/reactnative/builds/algoliasearch.reactnative.js]: Maybe this should have the same values as the swift client? https://github.com/algolia/algoliasearch-client-swift/blob/master/Source/AbstractClient.swift#L124

**Comment by bobylito** [comment]
I don't have answers for your question though :D

File: src/reactnative/builds/algoliasearch.reactnative.js (hunk 2 above)
Comment (ElPicador) [inline src/reactnative/builds/algoliasearch.reactnative.js]: It's 5 minutes on every other API clients. 
But maybe it'll make more sense to reduce it for the front-end implementations?

File: src/AlgoliaSearchCore.js (hunk 1 above)
Comment (vvo) [inline src/AlgoliaSearchCore.js]: Thanks for investigating still :) This is the actual SPEC.

File: src/reactnative/builds/algoliasearch.reactnative.js (hunk 2 above)
Comment (vvo) [inline src/reactnative/builds/algoliasearch.reactnative.js]: Yes, front-end facing API clients are meant to be used in situations with a low hitsPerPage. Not like a backend client, thus we can lower down the read timeout

**Comment by vvo** [comment]
New version of JS client is going out today 10am CET

**Pull request merged**